api/agent/agent.go

```diff
@@ -1221,6 +1198,7 @@ func newHotContainer(ctx context.Context, call *call, cfg *Config, id string, ud
 	}
 
 	return &container{
+		ctx:        ctx,
```

Comment:
nit: this is an anti-pattern (I know we do it in callHandle but we should fix that too), via https://golang.org/pkg/context/ : 

```
Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx
```

here I only see one usage for IsDone(), small change